test/sanitize_json.d

```diff
@@ -0,0 +1,231 @@
+import std.typecons : Flag, Yes, No;
+import std.array : Appender, appender;
+import std.bigint;
+import std.internal.cstring : tempCString;
+import std.string : startsWith, replace;
+import std.format : formattedWrite, format;
+import std.file : readText;
+import std.stdio;
+
+import json;
+
+__gshared File outFile;
+
+void usage()
+{
+    writeln("Usage: santize_json <input-json> [<output-json>]");
+}
+int main(string[] args)
+{
+    args = args[1 .. $];
+    if (args.length == 0)
+    {
+        usage();
+        return 1;
+    }
+    string inFilename = args[0];
```

Comment:
Why not use `stdin` if no arguments are provided?
If you don't like this, "-" is a common choice. If `-Xf` would support using stdout as output, this would allow to completely avoid the need of a temporary file.

Reply:
with stdin you can't use the file size to allocate the buffer to read the string into.   also by reading in a file you can use that filename in error messages :)

Reply:
> with stdin you can't use the file size to allocate the buffer to read the string into
just consume all stdin and that solves this problem; doesn't need to be processing input line by line for this use case.

> also by reading in a file you can use that filename in error messages :)
```
error: at __stdin.d:(12,10) { "foo": bar"} expected `"` got `b`
```

NOTE: __stdin.d is what is used in src/dmd/mars.d for same use case:
```
 if (strcmp(arguments[i + 1], "-") == 0)
*                        files.push("__stdin.d");
```